src/redis_lock/__init__.py

```diff
@@ -88,18 +100,37 @@ def id(self):
     def get_owner_id(self):
         return self._client.get(self._name)
 
-    def acquire(self, blocking=True):
+    def acquire(self, blocking=True, timeout=None):
+        """
+        :param blocking:
+            Boolean value specifying whether lock should be blocking or not.
+        :param timeout:
+            An integer value specifying the maximum number of seconds to block.
+        """
         logger.debug("Getting %r ...", self._name)
 
         if self._held:
             raise AlreadyAcquired("Already aquired from this Lock instance.")
 
+        if not blocking and timeout is not None:
+            raise TimeoutNotUsable
```

Comment:
I would add some text here, like `TimeoutNotUsable("Timeout cannot be used if blocking=True")`

Same for the other exceptions.